cachelib/cachebench/util/CacheConfig.h

```diff
@@ -41,6 +41,29 @@ class CacheMonitorFactory {
   virtual std::unique_ptr<CacheMonitor> create(Lru2QAllocator& cache) = 0;
 };
 
+// Parse memory tiers configuration from JSON config
+struct MemoryTierConfig : public JSONConfig {
+  MemoryTierConfig() {}
+
+  explicit MemoryTierConfig(const folly::dynamic& configJson);
+  MemoryTierCacheConfig getMemoryTierCacheConfig() {
+    MemoryTierCacheConfig config = memoryTierCacheConfigFromSource();
+    config.setRatio(ratio);
+    config.setMemBind(NumaBitMask(memBindNodes));
+    return config;
+  }
+
+  // Specifies ratio of this memory tier to other tiers
+  size_t ratio{0};
+  // Allocate memory only from specified NUMA nodes
+  std::string memBindNodes{""};
+
+private:
+  MemoryTierCacheConfig memoryTierCacheConfigFromSource() {
+    return MemoryTierCacheConfig::fromShm();
+  }
```

Comment:
Can you simplify this? by removing `memoryTierCacheConfigFromSource()` and hopefully `MemoryTierCacheConfig::fromShm()` as well. It looks enough to default construct `MemoryTierCacheConfig` in `getMemoryTierCacheConfig`.

For `fromShm()`, I think having code which is not used as of now just increases the complexity without value. We can add the code which implements the business logic when needed.

Reply:
Done